Samples/iOS-Benchmark/iOS-Benchmark/BenchmarkRunner.swift

```diff
@@ -0,0 +1,114 @@
+import Foundation
+import Benchmark
+import Capture
+
+private let kLogMessage = "50 characters long test message - 0123456789012345"
+
+final class BenchmarkRunner {
+    private(set) var runner: Benchmark.BenchmarkRunner
+
+    init(apiKey: String) {
+        let configurationSuite = BenchmarkSuite(name: "Configuration")
+        configurationSuite.register(benchmark: ConfigurationBenchmark(apiKey: apiKey))
+
+        let suites = [
+            configurationSuite,
+            kLogBenchmark,
+        ]
+
+        self.runner = Benchmark.BenchmarkRunner(
+            suites: suites,
+            settings: [],
+            customDefaults: defaultSettings
+        )
+    }
+
+    func run() {
+        try? self.runner.run()
+    }
+}
+
+final class ConfigurationBenchmark: AnyBenchmark {
+    private var logger: Logger?
+
+    private let apiKey: String
+
+    // MARK: AnyBenchmark
+
+    let name = "Configuration"
+    let settings: [Benchmark.BenchmarkSetting] = [Iterations(1)]
+
+    init(apiKey: String) {
+        self.apiKey = apiKey
+    }
+
+    func setUp() {}
+
+    func run(_: inout Benchmark.BenchmarkState) throws {
+        Logger.configure(
+            withAPIKey: self.apiKey,
+            sessionStrategy: .fixed()
+        )
+    }
+
+    func tearDown() {
+        // Give the Logger time to fetch the remove configuration and enter it's "normal" operation mode
+        // before moving to benchmarking logging performance.
+        Thread.sleep(forTimeInterval: 5.0)
+    }
+}
+
+private let kLogBenchmark = BenchmarkSuite(name: "Logging") { suite in
+    suite.benchmark("log without fields", settings: [WarmupIterations(10), Iterations(120)]) {
+        Logger.log(
+            level: .info,
+            message: kLogMessage,
+            file: nil,
+            line: nil,
+            function: nil,
+            fields: nil,
+            error: nil
+        )
+    }
+
+    suite.benchmark("log with 5 fields", settings: [WarmupIterations(10), Iterations(120)]) {
+        Logger.log(
+            level: .info,
+            message: kLogMessage,
+            file: nil,
+            line: nil,
+            function: nil,
+            fields: [
+                "keykeykey0": "valvalval0",
+                "keykeykey2": "valvalval1",
+                "keykeykey3": "valvalval2",
+                "keykeykey4": "valvalval3",
+                "keykeykey5": "valvalval4",
+            ],
+            error: nil
+        )
+    }
+
+    suite.benchmark("log with 10 fields", settings: [WarmupIterations(10), Iterations(120)]) {
```

Comment:
can we do more like 1k iterations? in all of them if these are in the order of ns 100s seem low